xayn-ai/src/ranker/public.rs

```diff
@@ -65,107 +58,35 @@ impl<SV, SM, KV, KM> Builder<SV, SM, KV, KM> {
-    /// to initialize the KPE.
-    ///
-    /// # Errors
-    ///
-    /// Fails if the data cannot be deserialized.
-    pub fn with_kpe_from_reader<W, N>(
-        self,
-        vocab: W,
-        bert: N,
-        cnn: impl Read,
-        classifier: impl Read,
-    ) -> Result<Builder<SV, SM, W, N>, Error> {
-        Ok(Builder {
-            smbert: self.smbert,
-            kpe: KPEBuilder::new(
-                vocab,
-                bert,
-                BinParams::deserialize_from(cnn)?,
-                BinParams::deserialize_from(classifier)?,
-            ),
-            user_interests: self.user_interests,
-        })
-    }
-
-    /// Sets the path of the vocabulary, bert model, CNN and classifier used
-    /// to initialize the KPE.
-    ///
-    /// # Errors
-    ///
-    /// Fails if a file cannot be opened.
-    pub fn with_kpe_from_file(
-        self,
-        vocab: impl AsRef<Path>,
-        bert: impl AsRef<Path>,
-        cnn: impl AsRef<Path>,
-        classifier: impl AsRef<Path>,
-    ) -> Result<Builder<SV, SM, impl BufRead, impl Read>, Error> {
-        Ok(Builder {
-            smbert: self.smbert,
-            kpe: KPEBuilder::from_files(vocab, bert, cnn, classifier)?,
-            user_interests: self.user_interests,
-        })
-    }
-
     /// Creates a `Reranker`.
     ///
     /// # Errors
     ///
     /// Fails if the SMBert or KPE cannot be initialized. For example because
     /// reading from a file failed or the bytes read are have an unexpected format.
-    pub fn build(self) -> Result<Ranker, Error>
-    where
-        SV: BufRead,
-        SM: Read,
-        KV: BufRead,
-        KM: Read,
-    {
-        let config = Configuration::default();
-        let smbert = SMBert::from(Arc::new(
-            self.smbert
-                .with_token_size(52)?
-                .with_accents(false)
-                .with_lowercase(true)
-                .with_pooling(AveragePooler)
-                .build()?,
-        ));
-        let coi = CoiSystem::new(config.clone(), smbert.clone());
-        let kpe = self
-            .kpe
+    pub fn build(self) -> Result<Ranker, Error> {
+        // TODO: (lj): Overriding passed-in configuration here seems somewhat surprising. Would it
+        //       make sense to pull these default out of the builder?
+        let smbert_config = self
+            .smbert_config
+            .with_token_size(52)?
+            .with_accents(false)
+            .with_lowercase(true)
+            .with_pooling(AveragePooler);
+
+        let smbert_pipeline = rubert::Pipeline::from(smbert_config)?;
+        let smbert = SMBert::from(Arc::new(smbert_pipeline));
+
+        let kpe_config = self
+            .kpe_config
             .with_token_size(150)?
             .with_accents(false)
-            .with_lowercase(false)
-            .build()?;
+            .with_lowercase(false);
 
+        let kpe = kpe::Pipeline::from(kpe_config)?;
```

Comment:
maybe
```rust
use kpe::Pipeline as KPE;

let kpe = KPE::from(kpe_config)?;
```
to make it look similar to SMBert

Reply:
Mmh, for me this only makes it harder to understand what's going on. Or in other words, why introduce an additional alias for a type, which the reader of the code then has to resolve?

Reply:
I personally think it is easier to read than 

```rust
kpe::Pipeline::from(kpe_config)
rubert::Pipeline::from(smbert_config);
rubert::Pipeline::from(qambert_config);
```

Reply:
As discussed, we'll leave it as is for now and think of better ways to name it than `Pipeline` in the future.